torch/csrc/distributed/rpc/tensorpipe_agent.cpp

```diff
@@ -19,10 +19,36 @@ namespace rpc {
 
 constexpr long kToMilliseconds = 1000;
 
+void TensorPipeAgent::collectNames() {
+  const worker_id_t selfId = workerInfo_.id_;
+  const std::string& selfName = workerInfo_.name_;
+
+  std::vector<uint8_t> selfNameVector(
+      (uint8_t*)selfName.c_str(),
+      (uint8_t*)selfName.c_str() + selfName.length());
+  addressStore_->set("names/" + std::to_string(selfId), selfNameVector);
+
+  workerIdToInfo_.emplace(selfId, WorkerInfo(selfName, selfId));
+  workerNameToInfo_.emplace(selfName, WorkerInfo(selfName, selfId));
+  for (worker_id_t workerId = 0; workerId < worldSize_; ++workerId) {
+    if (workerId == selfId) {
+      continue;
+    }
+    std::string workerKey = "names/" + std::to_string(workerId);
+    addressStore_->wait({workerKey});
+    std::vector<uint8_t> workerNameVector = addressStore_->get(workerKey);
+    std::string workerName(
+        (char*)workerNameVector.data(), workerNameVector.size());
+    workerIdToInfo_.emplace(workerId, WorkerInfo(workerName, workerId));
+    workerNameToInfo_.emplace(workerName, WorkerInfo(workerName, workerId));
+  }
+}
+
 TensorPipeAgent::TensorPipeAgent(
-    worker_id_t selfId,
-    std::string selfName,
     std::shared_ptr<::c10d::Store> addressStore,
+    std::string selfName,
+    worker_id_t selfId,
+    worker_id_t worldSize,
```

Comment:
I think we use `int` for `worldSize` instead of `worker_id_t`? Looks like worker_id_t is used only for ID's.